### PR TITLE
Fallback mechanism for missing IDN functions

### PR DESCRIPTION
**Pull request by annando**
See https://github.com/friendica/friendica/issues/14025#issuecomment-2041009659

### CODE REVIEW

**Comment by tobiasd** [comment]
Which Debian package should this be?

I had a look at 2023.12 and there it seems to be a composer dependency (from the lock file) but it was removed from there.

**Comment by annando** [comment]
It is `php-intl`.

File: doc/Install.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ Due to the large variety of operating systems and PHP platforms in existence we
 * Apache with mod-rewrite enabled and "Options All" so you can use a local `.htaccess` file
 * PHP 7.4+
   * PHP *command line* access with register_argc_argv set to true in the php.ini file
-  * Curl, GD, GMP, PDO, mbstrings, MySQLi, hash, xml, zip, IntlChar and OpenSSL extensions
+  * Curl, GD, GMP, PDO, mbstrings, MySQLi, hash, xml, zip, IntlChar, IDN and OpenSSL extensions
```
Comment (MrPetovan) [inline doc/Install.md:33]: This should read `intl` instead.

File: doc/de/Install.md [hunk 2]
```diff
@@ -27,7 +27,7 @@ Requirements
 * Apache mit einer aktiverten mod-rewrite-Funktion und dem Eintrag "Options All", so dass du die lokale .htaccess-Datei nutzen kannst
 * PHP  7.4+
   * PHP *Kommandozeilen*-Zugang mit register_argc_argv auf "true" gesetzt in der php.ini-Datei
-  * Curl, GD, GMP, PDO, mbstrings, MySQLi, hash, xml, zip, IntlChar and OpenSSL-Erweiterung
+  * Curl, GD, GMP, PDO, mbstrings, MySQLi, hash, xml, zip, IntlChar, IDN und OpenSSL-Erweiterung
```
Comment (MrPetovan) [inline doc/de/Install.md:30]: Hier sollte stattdessen `intl` stehen.

File: src/Core/Installer.php [hunk 3]
```diff
@@ -495,6 +495,13 @@ public function checkFunctions()
 		);
 		$returnVal = $returnVal ? $status : false;
 
+		$status = $this->checkFunction('idn_to_ascii',
+			DI::l10n()->t('IDN Functions PHP module'),
+			DI::l10n()->t('Error: IDN Functions PHP module required but not installed.'),
```
Comment (MrPetovan) [inline src/Core/Installer.php:500]: `IDN` -> `Intl`

File: src/Util/Network.php [hunk 4]
```diff
@@ -533,20 +533,29 @@ public static function convertToIdn(string $uri): string
...
 			}
 		}
 
 		return $uri;
 	}
 
+	private static function idnToAscii(string $uri): string
+	{
+		if (!function_exists('idn_to_ascii')) {
+			Logger::error('IDN functions are missing.');
```
Comment (MrPetovan) [inline src/Util/Network.php:553]: `IDN` -> `Intl`

**Pull request merged**